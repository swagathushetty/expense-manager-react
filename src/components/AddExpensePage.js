import React from 'react'
import { connect } from 'react-redux'
import ExpenseForm from './ExpenseForm'
import { startAddExpense} from '../actions/expenses'


export class AddExpensePage extends React.Component {
    onSubmit = (expense) => {
        //this is cannot be tested as addExpense is imported
        //we will abstract it
        // props.dispatch(addExpense(expense))
        this.props.startAddExpense(expense)

        //redirect to page without refrshing
        this.props.history.push('/')
    }
    render(){
        return (
            <div>
            <h1>Add expense</h1>
            <ExpenseForm
                onSubmit={this.onSubmit}
            />
            </div>
        )
    }
}


//we are doing this to make it testable
//we are sepearting the dispatch from components
//below works with dispatch
const mapDispatchToProps=(dispatch)=>{
    return{
        startAddExpense:(expense)=>dispatch(startAddExpense(expense))
    }

}

export default connect(undefined,mapDispatchToProps)(AddExpensePage)