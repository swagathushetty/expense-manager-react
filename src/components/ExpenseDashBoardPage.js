import React from 'react'
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'
import ExpensesSummary from './ExpensesSummary'

const ExpenseDasBoardPage = () => {
    return (
        <div>
             <ExpensesSummary />
            <ExpenseListFilters />
            <ExpenseList />
       </div>
    )
}

export default ExpenseDasBoardPage;
