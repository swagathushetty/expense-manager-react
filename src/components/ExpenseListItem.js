import React from 'react'
import {Link} from 'react-router-dom'
import {connect} from'react-redux'
import moment from 'moment'
import numeral from 'numeral'
numeral.locale('inr');

//before we used props as argument
//we did props.expenses.amount etc
export const ExpenseListItem=({id,description,amount,createdAt})=>{
    return (
        <div>
        <Link to={`/edit/${id}`}>
                <h3>{description}</h3>
        </Link>
           <p>
            {numeral(amount/100).format('$0,0.00')}
           -
           {moment(createdAt).format('MMMM Do,YYYY')}
           </p>
        </div>
    )
}

export default connect()(ExpenseListItem)

numeral.register('locale', 'inr', {
    delimiters: {
        thousands: ',',
        decimal: '.'
    },
    abbreviations: {
        thousand: 'k',
        million: 'm',
        billion: 'b',
        trillion: 't'
    },
    currency: {
        symbol: 'â‚¹'
    }
});
