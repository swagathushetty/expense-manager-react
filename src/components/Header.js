import React from 'react'
import { NavLink } from 'react-router-dom'

const Header = () => {

    return (<header>
        <h1>expensify</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}>Home page</NavLink>
        <NavLink to="/create" activeClassName="is-active"> expense page</NavLink>
        <NavLink to="/edit" activeClassName="is-active">Edit page</NavLink>
        <NavLink to="/help" activeClassName="is-active">Help page</NavLink>
    </header>
    )
}

export default Header