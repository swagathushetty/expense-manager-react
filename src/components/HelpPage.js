import React from 'react'

const HelpPage = () => {
    return (
        <div>
            this is from my help page component
       </div>
    )
}

export default HelpPage