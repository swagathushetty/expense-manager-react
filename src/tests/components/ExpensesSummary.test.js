import React from 'react'
import {shallow} from 'enzyme'
import {ExpensesSummary} from '../../components/ExpensesSummary'


test('should correctly render ExpensesSummary with 1 expense',()=>{
    const wrapper=shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />)
    expect(wrapper).toMatchSnapshot()
})

test('should correctly render ExpensesSummary with multiple', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={254543535} />)
    expect(wrapper).toMatchSnapshot()
})