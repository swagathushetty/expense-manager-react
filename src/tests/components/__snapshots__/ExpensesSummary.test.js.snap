// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    viewing
    1
     
    expense
     
    235
  </h1>
</div>
`;

exports[`should correctly render ExpensesSummary with 1 expense 2`] = `
<div>
  <h1>
    viewing
    23
     
    expenses
     
    254543535
  </h1>
</div>
`;
